src/Pact/Native/Ops.hs

```diff
@@ -224,18 +224,17 @@ coerceBinNum = binTy numA numA numA <> binTy tTyDecimal numA (numV "b")
 plusA :: Type n
 plusA = mkTyVar "a" [tTyString,TyList (mkTyVar "l" []),TySchema TyObject (mkSchemaVar "o") def]
 
-defTrunc :: NativeDefName -> Text -> (Decimal -> Integer) -> NativeDef
+defTrunc :: NativeDefName -> Text -> (Rational -> Integer) -> NativeDef
 defTrunc n desc op = defRNative n fun (funType tTyDecimal [("x",tTyDecimal),("prec",tTyInteger)] <>
                                       unaryTy tTyInteger tTyDecimal)
                      [ ExecExample $ "(" <> asString n <> " 3.5)"
                      , ExecExample $ "(" <> asString n <> " 100.15234 2)"
                      ]
                      (desc <> " value of decimal X as integer, or to PREC precision as decimal.")
     where fun :: RNativeFun e
-          fun _ [TLiteral (LDecimal d) _] = return $ toTerm $ op d
```

Comment:
The big surprise here is `truncate . roundTo' op 0` is much, much faster (~6us) than using a `RealFrac` methods on `Decimal` (~22us)

Reply:
Wow weird.